tests/test_mctpd.py

```diff
@@ -1279,6 +1279,35 @@ async def test_get_message_types(dbus, mctpd, routed_ep):
     rsp = await ep.send_control(mctpd.network.mctp_socket, cmd)
     assert rsp.hex(' ') == '00 04 00 01 f4 f3 f2 f1'
 
+""" Test GetMessageTypes when VDM is registered """
+async def test_get_message_types_with_vdm(dbus, mctpd, routed_ep):
+    ep = routed_ep
+
+    # Register VDM responder PCIe format
+    mctp = await mctpd_mctp_base_iface_obj(dbus)
+    v_type = asyncdbus.Variant('q', 0xABCD)
+    await mctp.call_register_vdm_type_support(0x00, v_type, 0x0001)
+
+    # Verify get message type response includes PCIe
+    cmd = MCTPControlCommand(True, 0, 0x05, bytes([0x00]))
+    rsp = await ep.send_control(mctpd.network.mctp_socket, cmd)
+    assert rsp.hex(' ') == '00 05 00 02 00 7e'
+
+    # Register VDM responder IANA format
+    v_type = asyncdbus.Variant('u', 0x1234ABCD)
+    await mctp.call_register_vdm_type_support(0x01, v_type, 0x5678)
+
+    # Verify get message type response includes IANA
+    cmd = MCTPControlCommand(True, 0, 0x05, bytes([0x00]))
+    rsp = await ep.send_control(mctpd.network.mctp_socket, cmd)
+    assert rsp.hex(' ') == '00 05 00 03 00 7e 7f'
+
+    await mctp.call_register_type_support(5, [0xF1F2F3F4])
+    # Verify get message type response includes SPDM and VDMs
+    cmd = MCTPControlCommand(True, 0, 0x05, bytes([0x00]))
+    rsp = await ep.send_control(mctpd.network.mctp_socket, cmd)
+    assert rsp.hex(' ') == '00 05 00 04 00 05 7e 7f'
+
```

Comment:
might be good to also test that the type is removed after unregistration, but you could also do that in the original unregistration test.